radixdlt/src/main/java/org/radix/universe/system/LocalSystem.java

```diff
@@ -83,10 +89,10 @@ int getJsonProcessors() {
 		return Runtime.getRuntime().availableProcessors();
 	}
 
-	public static LocalSystem create(SystemCounters counters, RuntimeProperties properties, Universe universe, String host) {
+	public static LocalSystem create(Supplier<Map<String, Object>> infoSupplier, RuntimeProperties properties, Universe universe, String host) {
```

Comment:
It would be much better from a documentation point of view, if we used a specific interface here (maybe `InfoSupplier`), rather than the generic interface `Supplier<Map<String, Object>>`.  As things are, it is impossible to determine what structure the map here should have, and the structure itself is kind of non-trivial.

The only way to determine what should be passed in here as an argument is to look up the callers, and inspect the non-trivial code there, which is more work than it should be.

A simple interface with some documentation should fix this easily.  Something like:
```
/**
 * Information supplier for debug information etc etc
 */
\@FunctionalInterface
public interface InfoSupplier {
    /**
     * Returns current values for debug information arranged etc and so forth...
     */
    Map<String, Object> getInfo();
}
```

Reply:
True, will do.

The reason why I keep these interfaces more "open-ended" is that I don't believe these are the correct interfaces we will use in the long run (need more info/design) and imo better to have clearly incorrect interfaces which are easily reversible over building out incorrect interfaces that ends up needing a bigger refactor.

Of course, where that line is I guess is a matter of opinion so will fix this